### PR TITLE
agent argument logic implement

### PR DESCRIPTION
**Pull request by xsa-dev**
_добавил предполагаемые параметры для запуска:_
- host,ho - строка: сервер для запуска мультиплеера по адресу передаваемому в строке
- join - строка: путь к серверу для запуска мультиплеера по адресу передаваемому в строке
- agent - строка: префикс к имени лог файла сохраняемом в папке с данными игры (__~/Library/Application Support/vcmi__)

### CODE REVIEW

**Comment by ShubusCorporation** [comment]
> 
> 
> _добавил предполагаемые параметры для запуска:_
> 
>     * host,ho - строка: сервер для запуска мультиплеера по адресу передаваемому в строке
> 
>     * join - строка: путь к серверу для запуска мультиплеера по адресу передаваемому в строке
> 
>     * agent - строка: префикс к имени лог файла сохраняемом в папке с данными игры (**~/Library/Application Support/vcmi**)

It's international project, use English, please.

File: client/CMT.cpp [hunk 1]
```diff
@@ -229,9 +235,20 @@ int main(int argc, char * argv[])
...
+	// agent name for different log files
+	std::string agent = "";
+	if (vm.count("agent")) {
+		agent = vm["agent"].as<std::string>();
+	}
+	std::string file = "VCMI_CLIENT_";
+	std::string ext = ".log";
+	std::string filename = file.append(agent);
+	filename = filename.append(ext);
+	//
```
Comment (ShubusCorporation) [inline client/CMT.cpp:249]: '//'  - What does it mean?! Without such comments, please.

Comment (ShubusCorporation) [inline client/CMT.cpp:249]: Assume, agent is empty. Last underscore is not needed char in this case.

File: client/CMT.cpp [hunk 2]
```diff
@@ -198,7 +198,13 @@ int main(int argc, char * argv[])
...
 		("serverport", po::value<si64>(), "override port specified in config file")
 		("saveprefix", po::value<std::string>(), "prefix for auto save files")
-		("savefrequency", po::value<si64>(), "limit auto save creation to each N days");
+		("savefrequency", po::value<si64>(), "limit auto save creation to each N days")
+		// 
+    ("host,ho", po::value<std::string>(), "create host game")
+    ("join,j", po::value<std::string>(), "join server, server:port")
+    ("agent,a", po::value<std::string>(), "agent name, agent name (log name)");
+	
+
```
Comment (ShubusCorporation) [inline client/CMT.cpp:207]: At least, align it properlyt. And only one empty line is enough.

File: client/CMT.cpp [hunk 3]
```diff
@@ -297,6 +314,11 @@ int main(int argc, char * argv[])
 	logConfig->configure();
 	logGlobal->debug("settings = %s", settings.toJsonNode().toJson());
 
+	// New attrs
```
Comment (ShubusCorporation) [inline client/CMT.cpp:317]: It will be new forever? No such comments, please.

File: client/CMT.cpp [hunk 4]
```diff
@@ -307,6 +329,7 @@ int main(int argc, char * argv[])
 		return false;
 	};
 
+
```
Comment (ShubusCorporation) [inline client/CMT.cpp:332]: Do not add extra empty lines, please

File: client/CMT.cpp (hunk 3 above)
Comment (nullkiller) [reply client/CMT.cpp:317]: If you need some comment for a part of code, you can move this code to a function as an option and use function name to express what comment does. This makes it readable and reduces size of parent function making it more readable as well.

Comment (nullkiller) [reply client/CMT.cpp:317]: Not sure if we should introduce a function here so at least consider making your comment more meaningful. Or removing it.

File: client/CMT.cpp (hunk 1 above)
Comment (nullkiller) [reply client/CMT.cpp:249]: The whole log section can be moved to a method setupClientLog() or something

**Comment by nullkiller** [comment]
Would be great to know a bit more wider idea behind this PR. What agent means here? Is it something multiplayer related?

**Comment by nullkiller** [comment]
I convert your pull request to draft until it is ready for review. Please convert it back when it is finished and ready for merge. Please avoid having too much work in a single pull request. Small PRs are much easier to review.

**Comment by nullkiller** [comment]
> > Would be great to know a bit more wider idea behind this PR. What agent means here? Is it something multiplayer related?
> 
> There are several basic ideas:
> 
> 1. Distribution of logs by agents to run several clients on one machine.
> 2. Adding arguments (the logic of their processing in development) to start a network game for the host and to join a multiplayer game from the command line.

Still not clear what agent is and what do you need it for. As far as I understood it is a process running headless server for multiplayer. Do you plan to implement some multiplayer functionality like lobby?

**Comment by nullkiller** [comment]
Stale PR, closing

**Pull request closed without merging**